### ISSUE TITLE
build errors with Pkg.add() 

### ISSUE THREAD

**Issue by amitmurthy** [JuliaParallel/MPI.jl#24]
With a Pkg.add("MPI") first I got this:

```
=====================================================================================[ ERROR: MPI ]======================================================================================

failed process: Process(`cmake -DCMAKE_INSTALL_PREFIX=/home/amitm/.julia/v0.4/MPI/deps/src -DCMAKE_LIB_INSTALL_PREFIX=/home/amitm/.julia/v0.4/MPI/deps/usr/lib ..`, ProcessExited(1)) [1]
while loading /home/amitm/.julia/v0.4/MPI/deps/build.jl, in expression starting on line 41
```

After I installed the devel package for openmpi separately from the console, doing a `Pkg.rm` and then trying again I got 

```
failed process: Process(`make`, ProcessExited(2)) [2]
while loading /home/amitm/.julia/v0.4/MPI/deps/build.jl, in expression starting on line 41
```

Doing a Pkg.rm("MPI") and using 
```
Pkg.clone("https://github.com/JuliaParallel/MPI.jl.git")
Pkg.build()
```
to install worked fine.

```
julia> versioninfo()
Julia Version 0.4.0-dev+2522
Commit 8cc787d (2015-01-06 05:27 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i5-4250U CPU @ 1.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by lcw** [comment]
I have decided not to install MPI if the user doesn't already have it installed.  I chose this because typically on cluster there is a specific version installed for the hardware.

It is a little hard to diagnose your problem with the information given.  For the case that failed can you provide the file `~/.julia/v0.4/MPI/deps/build/CMakeFiles`

**Comment by amitmurthy** [comment]
Unable to reproduce it now. Will reopen if I see it again.

**Comment by JaredCrean2** [comment]
I have encoutered the same problem.  The build completes but all the tests fails.  Here is the output from the test section:

```
Test project /home/dev1/.julia/v0.4/MPI/deps/build
    Start 1: test_bcast.jl_np:2
1/6 Test #1: test_bcast.jl_np:2 ...............***Failed    0.58 sec
    Start 2: test_reduce.jl_np:2
2/6 Test #2: test_reduce.jl_np:2 ..............***Failed    0.58 sec
    Start 3: test_sendrecv.jl_np:2
3/6 Test #3: test_sendrecv.jl_np:2 ............***Failed    0.58 sec
    Start 4: test_bcast.jl_np:5
4/6 Test #4: test_bcast.jl_np:5 ...............***Failed    0.75 sec
    Start 5: test_reduce.jl_np:5
5/6 Test #5: test_reduce.jl_np:5 ..............***Failed    0.82 sec
    Start 6: test_sendrecv.jl_np:5
6/6 Test #6: test_sendrecv.jl_np:5 ............***Failed    0.74 sec

0% tests passed, 6 tests failed out of 6

Total Test time (real) =   4.05 sec

The following tests FAILED:
	  1 - test_bcast.jl_np:2 (Failed)
	  2 - test_reduce.jl_np:2 (Failed)
	  3 - test_sendrecv.jl_np:2 (Failed)
	  4 - test_bcast.jl_np:5 (Failed)
	  5 - test_reduce.jl_np:5 (Failed)
	  6 - test_sendrecv.jl_np:5 (Failed)
Errors while running CTest
make: *** [test] Error 8
=================================[ ERROR: MPI ]=================================

LoadError: failed process: Process(`make test`, ProcessExited(2)) [2]
while loading /home/dev1/.julia/v0.4/MPI/deps/build.jl, in expression starting on line 41

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: MPI had build errors.

 - packages with build errors remain installed in /home/dev1/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("MPI")`
 - build a single package by running its `deps/build.jl` script

================================================================================
```
If I run the test script in the Readme file on Github, all processes give the error:
```
LoadError: UndefVarError: MPIManager not defined
```

Version info:
```
Julia Version 0.4.0-dev+2962
Commit 255e7ac* (2015-01-29 10:09 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Nehalem)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

Any ideas what is going on here?

**Comment by lcw** [comment]
I don't know.  @amitmurthy do you have any ideas?

**Comment by amitmurthy** [comment]
Since METADATA has not been updated, Pkg will refer to the last published version which did not include MPIManager. `Pkg.checkout("MPI")` should get you the latest, so that explains the `MPIManager not defined` error.

The regular tests do not use MPIManager, so it seems like a build issue. `cmake` is something I am totally unfamiliar with so cannot really help there. Sorry.

I'll update METADATA so that a `checkout` step is not required.

**Comment by JaredCrean2** [comment]
@lcw Looking around in the ```.julia/v0.4/MPI``` directory, I tried running the broadcast test manually based on the last few lines of the CMakeLists.txt file:
```
cd ~/.julia/v0.4/MPI/test
`/path/to/julia/exectuable ./run_test.jl ./test_bcast.jl
```
and got a warning aboutdeprecated syntax, but no error.
Did I run the test correctly, and if so, might the tests during building be treating the warning as an error?

**Comment by JaredCrean2** [comment]
After updating my PATH variable to include the version of julia I am trying to install MPI on (rather than a different version), the tests still fail during the build, but the test script in the Readme works.

Before updating my PATH, a copy of julia version 0.3 would be found, rather than the julia version 0.4 that was doing the executing the Pkg.build() command, so that likely caused some problems, but it doesn't explain why the tests failed during the build.

**Comment by amitmurthy** [comment]
OK. I am reopening this issue.

I have also updated METADATA and README to require the use of Julia 0.4 for MPIManager.

**Comment by lbenet** [comment]
Hi,
I'm having some problems to install MPI.jl (v0.2.6) on julia 0.4:
```julia
julia> versioninfo()
Julia Version 0.4.0-dev+3663
Commit a5b2a2f* (2015-03-03 20:03 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT NO_AFFINITY NEHALEM)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

Somehow it is not finding the open-mpi installed in my system (version 1.8.4, installed using homebrew), and keeps on looking for version 1.8.3:
```
-- Generating done
-- Build files have been written to: /Users/benet/.julia/v0.4/MPI/deps/build
[ 20%] Built target make_f_const
[ 20%] Built target version
[ 40%] Built target make_f2c
[ 60%] Generating mpi-build.jl
[ 60%] Built target mpijl-build
[ 80%] Generating compile-time.jl
[ 80%] Built target mpijl
make[2]: *** No rule to make target `/usr/local/Cellar/open-mpi/1.8.3/lib/libmpi_usempif08.dylib', needed by `libjuliampi.dylib'.  Stop.
make[1]: *** [CMakeFiles/juliampi.dir/all] Error 2
make: *** [all] Error 2
=================================[ ERROR: MPI ]=================================

LoadError: failed process: Process(`make`, ProcessExited(2)) [2]
while loading /Users/benet/.julia/v0.4/MPI/deps/build.jl, in expression starting on line 41

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: MPI had build errors.

 - packages with build errors remain installed in /Users/benet/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("MPI")`
 - build a single package by running its `deps/build.jl` script

================================================================================
```

Any ideas of what shall I change so it looks for my 1.8.4v libraries?

**Comment by lbenet** [comment]
Incidentally, `Pkg.build("MPI")` doesn't help...

**Comment by lcw** [comment]
@lbenet It is hard to tell what is going on.  I am guessing that homebrew is a little mixed up.  Probably some stuff left over after an upgrade.

Can you provide the full output of a fresh install of MPI.jl?

If you can also try the following and provide all of the output we might be able to figure out what is going on.  From your shell run

```
cd ~/julia/v0.4/MPI/deps
git clean -x -d -f
mkdir build
cd build
cmake -LAH ..
```

**Comment by lbenet** [comment]
@lcw  I followed your suggestion, reinstalling first MPI.jl, which lead to the same problem I reported. So, I  cleaned and rebuilt as you suggested. The first time it found ifort (instead of gfortran), but everything went fine, so I set `FC` to point to gfortran, cleaned and rebuit again. No problem...

So I removed MPI.jl and re-add it, and now everything seems to have worked out nicely. 

Probably it was some confusion due to homebrew. Thanks for the help!
